Content.Shared/Humanoid/SharedHumanoidAppearanceSystem.cs

```diff
@@ -132,6 +132,38 @@ public void SetLayerVisibility(EntityUid uid,
             Dirty(uid, humanoid);
     }
 
+    /// <summary>
+    ///     Clones a humanoid's appearance to a target mob, provided they both have humanoid components.
+    /// </summary>
+    /// <param name="source">Source entity to fetch the original appearance from.</param>
+    /// <param name="target">Target entity to apply the source entity's appearance to.</param>
+    /// <param name="sourceHumanoid">Source entity's humanoid component.</param>
+    /// <param name="targetHumanoid">Target entity's humanoid component.</param>
+    public void CloneAppearance(EntityUid source, EntityUid target, HumanoidAppearanceComponent? sourceHumanoid = null,
+        HumanoidAppearanceComponent? targetHumanoid = null)
+    {
+        if (!Resolve(source, ref sourceHumanoid) || !Resolve(target, ref targetHumanoid))
+        {
+            return;
+        }
+
+        targetHumanoid.Species = sourceHumanoid.Species;
```

Comment:
Obviously just a translation of the server code, but yeah, feels like this could be more maintainably expressed somehow.

Reply:
copy comp probably (wait for Engine PR approval)